src/deepsparse/v2/operators/operator.py

```diff
@@ -90,21 +93,39 @@ def __call__(
         else:
             inference_input = kwargs
 
-        run_output = self.run(inp=inference_input, context=context)
-
-        if self.has_output_schema():
-            return self.output_schema(**run_output)
-        return run_output
+        run_output, state_update = self.run(
+            inp=inference_input,
+            context=context,
+            pipeline_state=pipeline_state,
+            inference_state=inference_state,
+        )
+        if self.has_output_schema() and not isinstance(run_output, self.output_schema):
+            return self.output_schema(**run_output), state_update
+        return run_output, state_update
 
     @abstractmethod
-    def run(self, inp: Any, context: Optional[Context]) -> Any:
+    def run(
+        self,
+        inp: Any,
+        context: Optional[Context],
+        pipeline_state: PipelineState,
+        inference_state: InferenceState,
+    ) -> Any:
         """
         :param inp: operator input, as the defined input schema if applicable
         :param context: pipeline context of already run operators
         :return: result of this operator as the defined output schema if applicable
         """
         raise NotImplementedError
 
+    def can_operate(
+        self, inp: Any, context: Context, inference_state: InferenceState
+    ) -> bool:
+        """
+        Whether or not the given operator can run, based on input, context, or state
+        """
+        raise NotImplementedError
```

Comment:
not sure if we should require this to be implemented, any reason to not default to `True` or some basic default implementation?

Reply:
yes should probably just be true by default.